Library/Homebrew/download_strategy.rb

```diff
@@ -184,7 +185,19 @@ def stage
       with_system_path { buffered_write("bunzip2") }
     when :gzip, :bzip2, :compress, :tar
       # Assume these are also tarred
-      with_system_path { safe_system "tar", "xf", cached_location }
+      if @extracts.empty?
+        with_system_path { safe_system "tar", "xf", cached_location }
+      else
+        @extracts.each do |item|
+          extract_path = `tar -t --include "*#{item}" -f "#{cached_location}"`
```

Comment:
Ah, OK. I was struggling fruitlessly to find any way to get something other than "true" or "false" out of safe_system. An alternative might be to use some variant of pipe_to_tar or zcat + a pipe_to_pax. (The simplest answer, of course, is to just require the relative path within the tar file in the patch do block, so that it and its depth don't have to be computed, or the uniqueness verified, but that seemed like it would make it more annoying to formula writers to use the extracts key overall.)